components/widgets/inputs/dropdown_input_hybrid.tsx

```diff
@@ -0,0 +1,253 @@
+            selected: props.isSelected,
+            focused: props.isFocused,
+        })}
+    >
+        <components.Option {...props}/>
+    </div>
+);
+
+const DropdownInputHybrid = <T extends ValueType>(props: Props<T>) => {
+    const {
+        value,
+        placeholder,
+        className,
+        name,
+        legend, 
+        onDropdownChange,
+        onInputChange,
+        error,
+        exceptionToInput,
+        width,
+        inputValue,
+        inputType,
+        defaultValue,
+        ...otherProps
+    } = props;
+
+    const containerRef = useRef<HTMLInputElement>(null);
+    const inputRef = useRef<HTMLInputElement>(null);
+    const [inputFocused, setInputFocused] = useState(false);
+    const [focused, setFocused] = useState(false);
+    const [showInput, setShowInput] = useState(!!inputValue);
+
+    useEffect(() => {
+        if (showInput && !inputValue && inputRef.current) {
+            inputRef.current.focus();
+        }
+    }, [showInput]);
+
+    useEffect(() => {
+        if (!inputFocused) {
+            showTextInput(inputValue, false);
+        }
+    }, [inputValue]);
+
+    useEffect(() => {
+        if (!inputValue && !focused && !inputFocused) {
+            onDropdownChange(props.defaultValue);
+            showTextInput('');
+        }
+    }, [focused, inputFocused]);
+
+    const getMenuStyles = () => {
+        if (showInput) {
+            return {
+                menu: (provided: CSSProperties) => ({
+                    ...provided,
+                    width: containerRef.current ? `${containerRef.current.offsetWidth}px` : '0px',
+                    left: inputRef.current ? `-${inputRef.current.offsetWidth}px` : '0px',
+                }),
+            };
+        }
+        return {};
+    }
+
+    const onInputBlur = () => setInputFocused(false);
+
+    const onInputFocus = () => setInputFocused(true);;
+
+    const onDropdownInputFocus = (event: React.FocusEvent<HTMLElement>) => {
+        const {onFocus} = props;
+        
+        setFocused(true);
+
+        if (onFocus) {
+            onFocus(event);
+        }
+    };
+
+    const onDropdownInputBlur = (event: React.FocusEvent<HTMLElement>) => {
+        const {onBlur} = props;
+
+        setFocused(false);
+
+        if (onBlur) {
+            onBlur(event);
+        }
+    };
+
+    const onValueChange = async (event: T) => {
+        const {onDropdownChange} = props;
+
+        showTextInput(event.value);
+
+        if (onDropdownChange) {
+            onDropdownChange(event);
+        }
+    };
+    
+    // We want to show the text input when we have a dropdown value selected and 
+    const showTextInput = (val: string, focus = true) => {
+        if (!val || exceptionToInput.includes(val)) {
+            setShowInput(false);
+        } else {
+            setShowInput(true);
+            if (inputRef.current && focus) {
+                inputRef.current.focus();
+            }
```

Comment:
nit: could be simpler
```suggestion
            if (focus) {
                inputRef.current?.focus();
            }
```